infrastructure/modules/worker_with_sidecar/main.tf

```diff
@@ -10,6 +10,9 @@ module "service" {
   desired_task_count             = var.desired_task_count
   security_group_ids             = var.security_group_ids
   use_fargate_spot               = var.use_fargate_spot
+  container_name                 = var.container_name
```

Comment:
I would create a specific "api" module following the pattern in storage: https://github.com/wellcomecollection/storage-service/tree/master/terraform/modules/service/api

Reply:
My instinct was to abstract away common functionality into one of these workers so that there isn't loads of duplication, but do you think that isn't such an issue with our terraform?

Also, I don't know if creating a new module is necessary, couldn't we just stick everything in the [current catalogue API module](https://github.com/wellcomecollection/catalogue/tree/master/api/terraform/modules/catalogue_api_stack) seeing as thats the only place it is going to be used for now?

Reply:
I think using another module with some duplication makes it clearer what makes up the service. I use the term "worker" to refer to a queue backed service, so it feels wrong.

> Also, I don't know if creating a new module is necessary, couldn't we just stick everything in the current catalogue API module seeing as thats the only place it is going to be used for now?

That would work.

My preference would be to create a new module anyway as part of the process of replacing the old one: Create new module -> create an instance of the new module -> replace old running instance with new instance -> delete old instance & module.